### PR TITLE
Fixes #10222: Add guideline for technique naming

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/10222

### CODE REVIEW

File: 8_reference/73_techniques_best_practices.txt [hunk 1]
```diff
@@ -25,6 +25,18 @@
...
 ==== Writing convention
 
+===== Technique Naming Guidelines
+
+The following rules should be followed when naming a new Technique:
+
+   * Try to keep names as short as possible, to improve readability
+   * Read the exinsting technique list, and particularly techniques related to what you are writing. The new names should be consistent with existing ones.
+   * The name should be a nominal group. It describes the target of the action, not the action itself.
+   * The general name should like like: General Concept (package, file, etc.) + Source (from file, etc.) + Implementation details (platform, software name, etc.)
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: like like? look like?

File: 8_reference/73_techniques_best_practices.txt [hunk 2]
```diff
@@ -25,6 +25,18 @@
...
+   * The general idea it to go from the most general information to the most precise.
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: s/it/is/

**Comment by amousset** [comment]
Commit modified

File: 8_reference/73_techniques_best_practices.txt [hunk 3]
```diff
@@ -25,6 +25,18 @@
 
 ==== Writing convention
 
+===== Technique Naming Guidelines
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: Just like your guidelines say for technique names, I think this should not be capitalised ;)

File: 8_reference/73_techniques_best_practices.txt [hunk 4]
```diff
@@ -25,6 +25,18 @@
...
+
+The following rules should be followed when naming a new Technique:
+
+   * Try to keep names as short as possible, to improve readability
+   * Read the exinsting technique list, and particularly techniques related to what you are writing. The new names should be consistent with existing ones.
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: s/exinsting/existing/

File: 8_reference/73_techniques_best_practices.txt [hunk 5]
```diff
@@ -25,6 +25,18 @@
...
+   * The name should be a nominal group (not "Manage file content" or "Set Service state" but "File content" or "Service state)". It describes the target of the action, not the action itself.
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: Always start with good examples to have bad examples later. Also, the phrasing here doesn't make it very clear what is good or bad.

File: 8_reference/73_techniques_best_practices.txt [hunk 6]
```diff
@@ -25,6 +25,18 @@
...
+   * The name should look like: General Concept (package, file, etc.) + Source (from file, etc.) + Implementation details (platform, software name, etc.)
```
Comment (jooooooon) [inline 8_reference/73_techniques_best_practices.txt]: Examples of this would be useful too. Package repositories may be a good example, or variables?

**Comment by amousset** [comment]
Commit modified

**Comment by jooooooon** [comment]
👍

**Pull request merged**